### PR TITLE
Update .NET Monitor readmes

### LINKED ISSUE #5933: .NET Monitor docs should describe purpose and usage of dotnet/monitor/base image
There are two .NET Monitor images that are produced:
- `dotnet/monitor`
- `dotnet/monitor/base`

It is not obvious what the difference is between these two images and why a customer would choose one over the other.

Brief explanation of the difference between the two:
- `dotnet/monitor` is the image intended for the vast majority of customers to use. It is the entire composite of the full .NET Monitor offering, including the base tool installation and all .NET Monitor extensions.
- `dotnet/monitor/base` is a base image that contains only the core .NET Monitor package. Extensions such as egress providers do not exist in this image. This image is meant as a building block for producing a customized image that only contains the extensions that a given customer would need.

Fortunately, the featured repos of most of the documentation steer customers toward the `dotnet/monitor` image. However, the `dotnet/monitor/base` image has very little documentation about why it exists and how it should be used.

The current proposal is that:
- the `dotnet/monitor` documentation should be updated to succinctly refer to the base image and its purpose
- the `dotnet/monitor/base` documentation should be updated to describe in detail the purpose of the image and how it can be used.

### PR DESCRIPTION
**Pull request by jander-msft**
These changes to the .NET Monitor readmes are intended to better describe the intended usage of base and derived .NET Monitor images and link to more examples of how the derived image is intended to be used. The `dotnet/monitor/base` readme attempts to steer customers to the `dotnet/monitor` readme for those who are intending to use the full functionality of .NET Monitor rather than building a customized .NET Monitor image.

closes #5933

### CODE REVIEW

File: README.monitor-base.md [hunk 1]
```diff
@@ -9,9 +9,9 @@
 
 ## About
 
-This image contains the .NET Monitor Base installation.
+This image contains the base installation of .NET Monitor, a diagnostic tool for capturing diagnostic artifacts (such as dumps and traces) in an operator-driven or automated manner. This tool is an ASP.NET application that hosts a web API for inspecting .NET processes and collecting diagnostic artifacts.
 
-Use this image as a base image for building a .NET Monitor image with extensions.
+This image only provides the base functionality of the .NET Monitor tool; it is only meant to be used as a base image upon which .NET Monitor extensions are installed. If you are looking for the full feature set that is provided by the .NET Monitor global tool (including the egress capabilities), see the [dotnet/monitor](https://github.com/dotnet/dotnet-docker/blob/main/README.monitor.md) image.
```
[the remark this answers is not in the document]
Comment (jander-msft) [reply README.monitor-base.md:14]: This link is dynamically generated for each type of readme (DH, MAR, GH) from the About.monitor-base.md template. So I think I either leave it as or special case the GH link to be locally referring. Do you want the latter (which is effectively your suggestion) or to leave it as-is?

[the remark this answers is not in the document]
Comment (jander-msft) [reply README.monitor-base.md:14]: Sorry, I missed your opening sentence and went straight to the suggestion. I'll update the template to special case the GH readme.

**Comment by jander-msft** [comment]
fyi @dotnet/dotnet-monitor

**Comment by lbussell** [comment]
@jander-msft do we need to wait for anyone from Monitor team to review this?

**Comment by jander-msft** [comment]
> @jander-msft do we need to wait for anyone from Monitor team to review this?

I think we're good to go. I'll follow up with another PR if there are any requested changes.

**Pull request merged**